App/views.py

```diff
@@ -0,0 +1,26 @@
+from django.http import HttpResponse
+from django.shortcuts import render
+import json
+import requests
+
+def main(request):
+    if(request.method=='POST'):
+        usr = request.POST['username']  #Here, we get the username fetched from html page
+        usr = usr.replace("&","%26")
+        URL = "https://en.wikipedia.org/w/api.php?action=query&format=json&list=usercontribs&ucuser="
+        URL = URL + usr
+        r = requests.get(URL)  #Recieved data in json-format
+        if(r.status_code != 200):
+            return(render(request,'App/index.html',{"error":'HTTP response ' + str(r.status_code)}))
+        else:
+            lst = json.loads(r.text)
+            try:    
+                if(lst['error']):
+                    return(render(request,'App/index.html',{"error":'INVALID INPUT'}))
```

Comment:
There might be other reasons for getting an API. You probably want to show the user the error message returned by the API instead.

Reply:
Done.